File: PGS/Part04/SoHyun/L3_150365.py
def dist(x, y, r, c):
    return abs(x - r) + abs(y - c)

def solution(n, m, x, y, r, c, k):
    if (k - dist(x, y, r, c)) % 2 or k < dist(x, y, r, c):
        return "impossible"
    answer = ""
    move = 0
    
    while x < n and (k - move) > dist(x, y, r, c):
        move += 1
        x += 1
        answer += "d"
        
    while 1 < y and (k - move) > dist(x, y, r, c):
        move += 1
        y -= 1
        answer += "l"
    
    while (k - move) > dist(x, y, r, c):
        move += 2
        answer += "rl"
        
    if x < r:
        answer += "d" * (r - x)
        x = r
    if y > c:
        answer += "l" * (y - c)
        y = c
    if y < c:
        answer += "r" * (c - y)
        y = c
    if x > r:
        answer += "u" * (x - r)
        x = r
    
    return answer

File: PGS/Part04/JaeHyeon/L3_150365.java
import java.util.*;

class Solution {
    int[] dr = new int[]{1, 0, 0, -1};
    int[] dc = new int[]{0, -1, 1, 0};
    char[] cmd = new char[]{'d', 'l', 'r', 'u'};
    boolean isFound = false;
    String answer;
    public String solution(int n, int m, int x, int y, int r, int c, int k) {

        dfs(n, m, x - 1, y - 1, r - 1, c - 1, k, 0, new StringBuilder());
        
        return isFound ? answer : "impossible";
    }
    
    void dfs (int n, int m, int x, int y, int r, int c, int k, int depth, StringBuilder sb) {
        if (isFound)    return;
        
        int remain = k - depth;
        int dist   = Math.abs(x - r) + Math.abs(y - c);
        if (dist > remain || (remain - dist) % 2 == 1) return;
        
        if (depth == k) {
            if (x == r && y == c) {
                answer = sb.toString();
                isFound = true;
            }
            return;
        }
        
        for (int d = 0; d < 4; d++) {
            int nr = x + dr[d];
            int nc = y + dc[d];
            
            if (nr >= 0 && nr < n && nc >= 0 && nc < m) {
                sb.append(cmd[d]);
                dfs(n, m, nr, nc, r, c, k, depth + 1, sb);
                sb.deleteCharAt(sb.length() - 1);
            }
        }
    }
}